pandas/tests/groupby/test_rank.py

```diff
@@ -472,11 +470,23 @@ def test_rank_avg_even_vals(dtype, upper):
 )
 def test_rank_object_raises(ties_method, ascending, na_option, pct, vals):
     df = DataFrame({"key": ["foo"] * 5, "val": vals})
+    mask = df["val"].isna()
 
-    with pytest.raises(DataError, match="No numeric types to aggregate"):
-        df.groupby("key").rank(
-            method=ties_method, ascending=ascending, na_option=na_option, pct=pct
-        )
+    gb = df.groupby("key")
+    res = gb.rank(method=ties_method, ascending=ascending, na_option=na_option, pct=pct)
+
+    # construct our expected by using numeric values with the same ordering
+    if mask.any():
+        df2 = DataFrame({"key": ["foo"] * 5, "val": [0, np.nan, 2, np.nan, 1]})
+    else:
+        df2 = DataFrame({"key": ["foo"] * 5, "val": [0, 0, 2, 0, 1]})
```

Comment:
putting this in the parameters means it gets constructed at test collection time, increasing the tests' memory footprint, so i try to avoid that pattern.

i also think this makes it more obvious where df2 comes from than if it were in the parametrization